### ISSUE TITLE
Don't show welcome message when `--getProperty` was passed

### ISSUE THREAD

**Issue by rotanov** [dotnet/sdk#38444]
When a user executes `dotnet build` command with `--getProperty`, for example:

```
dotnet build -c <configuration> --getProperty:OutputPath --getProperty:OutputType --getProperty:AssemblyName <project>
```

Most likely that it's output is gonna be parsed.

However in some cases (or every time in CI, in clean docker environment) the welcome message is being prepended to the output, which breaks parsing.

I've seen https://github.com/dotnet/sdk/issues/3828 and https://github.com/dotnet/cli/pull/13275 but it seems to me in this case it's clear that user never wants a welcome message, so can you consider never showing welcome message if `--getProperty` was passed?

Also, bonus suggestion, may be everyone will benefit if welcome message is only displayed when `dotnet` (or `dotnet build` in case welcome message is tied only with build command) is executed with no arguments?

**Comment by baronfel** [comment]
This will be more difficult to do than folks expect - the SDK's understanding of MSBuild properties (like -getProperty) is limited, and the MSBuild parsing logic happens much further along than the welcome message. We'd have to do some serious reshuffling. More broadly we have requests to look at how the SDK can spew text to stdout, and we think the more comprehensive approach would be to make out-of-band information like the welcome message write to stderr instead of stdout. This will require some finess, though, because until recently Powershell interpreted anything written to stderr as an actual error.

**Comment by rotanov** [comment]
> because until recently Powershell interpreted anything written to stderr as an actual error
Yeah, if that's the case, it will fail CI job

**Comment by rotanov** [comment]
> This will be more difficult to do than folks expect - the SDK's understanding of MSBuild properties (like -getProperty) is limited

If welcome message is SDK's thing and not a `dotnet build` thing, then what about my "bonus suggestion" i.e. to only show welcome message when no arguments are passed to `dotnet`? Does your comment stand still in that case?

**Comment by rotanov** [comment]
Update:
We tried following workarounds:

1. `DOTNET_NOLOGO=1` doesn't work 

Next, we decided to try running some dotnet command which will trigger welcome message first (in `before_script:` section of gitlab ci), so actually useful commands in the job won't display it:

2. `dotnet` doesn't trigger welcome message
3. `dotnet build` DOES trigger welcome message, but returns non zero exit code and fails the job
4. `dotnet --info` doesn't trigger welcome message

So, in other words, commands which I expect to display welcome message are not showing it, and commands which I expect to do the job do show the welcome message.
